addons/mrp/wizard/mrp_markdone_confirmation.py

```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class MrpMarkdoneConfirmation(models.TransientModel):
+    _name = 'mrp.markdone.confirmation'
+
+    @api.multi
+    def action_done(self):
+        self.ensure_one()
+        return self.env['mrp.production'].browse(self._context.get('active_id')).action_done()
```

Comment:
Instead of using `active_id` like this create a m2o on this model called `production_id` and initialize that to the correct manufacturing order when creating the wizard in `button_mark_done`. In there you can pass the id of the wizard you create in `res_id`.